misc/nix/shell.nix

```diff
@@ -10,33 +10,30 @@
 { pkgs ? import <nixpkgs> {} }:
 with pkgs;
 
-let
-    target-analyzer = builtins.toString ./../../.rust-analyzer/target;
-    scoped-rust-analyzer = pkgs.writeShellScriptBin "rust-analyzer" ''
-      exec ${pkgs.coreutils}/bin/env CARGO_TARGET_DIR=${target-analyzer} ${pkgs.rust-analyzer}/bin/rust-analyzer
-    '';
-in
 stdenv.mkDerivation rec {
   name = "materialize";
   buildInputs = with pkgs; [
-      clang_14
-      cmake
-      rustup
-      openssl
-      postgresql
-      pkg-config
-      lld_14
-      python311
-      scoped-rust-analyzer
-      shellcheck
-  ];
+    libclang
+    rustPlatform.bindgenHook
+    cmake
+    perl
+    rustup
+    pkg-config
+    python3
+    openssl
 
-  hardeningDisable = [ "fortify" ];
+    postgresql
+    git
 
-  RUSTFLAGS = "-Clinker=clang -Clink-arg=--ld-path=${pkgs.mold}/bin/mold -Clink-arg=-Wl,--warn-unresolved-symbols -Cdebuginfo=1 -Csymbol-mangling-version=v0 --cfg=tokio_usntable";
+    # For lint checks
+    shellcheck
+    buf
+    nodejs_22
+  ] ++ lib.optionals stdenv.isDarwin [
+    libiconv
+    darwin.apple_sdk.frameworks.DiskArbitration
+    darwin.apple_sdk.frameworks.Foundation
+  ];
 
-  shellHook = ''
-    export LD_LIBRARY_PATH=${pkgs.stdenv.cc.cc.lib}/lib:$LD_LIBRARY_PATH
-    export LIBCLANG_PATH="${llvmPackages_14.libclang.lib}/lib";
-  '';
+  RUSTFLAGS = "-Cdebuginfo=1 -Csymbol-mangling-version=v0";
 }
```

Comment:
These changes seem material. Can you check with @petrosagg that they won't break his workflow? 

Then again the `--cfg=tokio-usntable` typo doesn't exactly inspire confidence that was was here was working well...

Reply:
I'll try this out but is there a reason we're removing `mold` as the linker?

Reply:
`mold` doesn't work on macOS, so I had made it conditional for Linux only at first, but even there I was getting errors. I can take another look if we want to keep mold around on Linux only.

Same for the other flags, which I removed one by one until the shell worked.

Reply:
I looked into this today I get some linking errors when optionally enabling `mold` for Linux. I'll have to check which of the changes breaks it. The failure I got was failing to find `Scrt1.o`

Reply:
Does the original code work for you? `bin/environmentd` fails for me while building `rdkafka-sys` with openssl related errors.

Reply:
It works, yeah. I only have issues with building the `openssl` crate when it's built in non-vendored mode but we do have the `vendored` feature enabled in MZ. Maybe that is ignored on OSX?

Reply:
The new code here also works for me, I tried a build from scratch, I just wanted to keep the mold option for linux

Reply:
> It works, yeah. I only have issues with building the `openssl` crate when it's built in non-vendored mode but we do have the `vendored` feature enabled in MZ. Maybe that is ignored on OSX?

Ah okay. I tried on Linux actually. With the original shell.nix, I get:
```bash
$ nix-shell misc/nix/shell.nix
$ bin/environmentd
  Building and linking librdkafka statically
  CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 3.5 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.

  CMake Error at src/CMakeLists.txt:235 (get_target_property):
    get_target_property() called with non-existent target "OpenSSL::SSL".

  thread 'main' panicked at /home/artimaeus/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5:
```
Am I missing something?

Reply:
Tested on both macOS and Linux. `nix-shell --pure misc/nix/shell.nix` successfully runs `bin/environmentd`.